aikau/src/test/resources/alfresco/documentlibrary/SearchListTest.js

```diff
@@ -75,79 +75,68 @@ define(["intern!object",
             .click()
          .end()
          // Check that updating the hash results in a search request being made...
-         .findAllByCssSelector(TestCommon.pubSubDataCssSelector("last", "term", "testTerm1"))
-            .then(function(elements) {
-               assert(elements.length === 1, "Search term didn't request search");
-            });
-      },
-
-      "Test that facet fields are included in search": function() {
-         // Make sure our facet was included...
-         return browser.findAllByCssSelector(TestCommon.pubSubDataCssSelector("last", "facetFields", "qname1"))
-            .then(function(elements) {
-               assert(elements.length === 1, "Facet fields not set appropriately from request");
-            });
+         .getLastPublish("ALF_SEARCH_REQUEST")
+         .then(function(payload) {
+            assert.equal(payload.term, "testTerm1", "Search term didn't request search");
+            assert.equal(payload.facetFields, "qname1", "Facet fields not set appropriately from request");
```

Comment:
Can also do `assert.propertyVal(payload, "facetFields", "qname1", "Facet fields...");`
Deep testing also available with `assert.deepPropertyVal(payload, "foo.bar.wibble", "expectedValue", "Value not correct");`
There are also corresponding negative methods on the `assert` object of `propertyNotVal()` and `deepPropertyNotVal()` with similar syntax.
But this works too, of course :)